restcontrollers/assetController.go

```diff
@@ -12,214 +11,194 @@ func (c *RESTController) updateUserSettings(w http.ResponseWriter, r *http.Reque
 	log.Println("Update user settings")
 	data, err := decodePostData(w, r)
 	if err != nil {
+		writeError(err.Error(), w, http.StatusBadRequest)
 		return
 	}
 
 	userData, err := parseUserData(data)
 	if err != nil {
-		w.WriteHeader(http.StatusBadRequest)
-		fmt.Fprint(w, "Failed to get suer")
+		writeError(err.Error(), w, http.StatusBadRequest)
+		return
 	}
 
 	err = c.DBController.UpdateUserSettings(userData)
 	if err == nil {
```

Comment:
Try to use early return instead.
Check for errors, handle errors, and go on with this part.